src/OrleansAzureUtils/Hosting/AzureSilo.cs

```diff
@@ -174,6 +169,8 @@ public bool Start(ClusterConfiguration config, string deploymentId = null, strin
 
             // Bootstrap this Orleans silo instance
 
+            var deploymentId = host.Config.Globals.DeploymentId;
```

Comment:
yeah, I also find this very confusing, and think it stems from when Orleans was XML config based only. I think that even today there are still deployments with a static XML configuration, but they want to just override the deploymentId in code. Granted, instead of just calling Start, they could go through the process of manually loading the XML config, then updating the deploymentId, and then calling start.
But I'd be wary about making this breaking change, when we are already thinking that several breaking changes to config/startup are coming in our Orleans 2.0 release.